#
# For slabtop to work properly, the kernel's slab names must not
# contain any spaces; see discussion recently on LKML
# (http://marc.theaimsgroup.com/?l=linux-kernel&m=106420809501130&w=2).
#
# This patch includes a patch against kernel 2.4.21 to fix slab names
# with spaces;
#
--- linux-2.4.21/fs/dnotify.c.orig	2003-09-22 12:21:27.000000000 -0700
+++ linux-2.4.21/fs/dnotify.c	2003-09-22 12:21:38.000000000 -0700
@@ -131,7 +131,7 @@
 
 static int __init dnotify_init(void)
 {
-	dn_cache = kmem_cache_create("dnotify cache",
+	dn_cache = kmem_cache_create("dnotify_cache",
 		sizeof(struct dnotify_struct), 0, 0, NULL, NULL);
 	if (!dn_cache)
 		panic("cannot create dnotify slab cache");
--- linux-2.4.21/fs/locks.c.orig	2003-09-22 12:22:45.000000000 -0700
+++ linux-2.4.21/fs/locks.c	2003-09-22 12:23:10.000000000 -0700
@@ -1940,7 +1940,7 @@
 
 static int __init filelock_init(void)
 {
-	filelock_cache = kmem_cache_create("file lock cache",
+	filelock_cache = kmem_cache_create("file_lock_cache",
 			sizeof(struct file_lock), 0, 0, init_once, NULL);
 	if (!filelock_cache)
 		panic("cannot create file lock slab cache");
--- linux-2.4.21/fs/fcntl.c.orig	2003-09-22 12:23:41.000000000 -0700
+++ linux-2.4.21/fs/fcntl.c	2003-09-22 12:24:02.000000000 -0700
@@ -525,7 +525,7 @@
 
 static int __init fasync_init(void)
 {
-	fasync_cache = kmem_cache_create("fasync cache",
+	fasync_cache = kmem_cache_create("fasync_cache",
 		sizeof(struct fasync_struct), 0, 0, NULL, NULL);
 	if (!fasync_cache)
 		panic("cannot create fasync slab cache");
